seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java

```diff
@@ -369,30 +477,59 @@ private void restoreJobFromMasterActiveSwitch(@NonNull Long jobId, @NonNull JobI
         }
 
         String jobFullName = jobMaster.getPhysicalPlan().getJobFullName();
-        runningJobMasterMap.put(jobId, jobMaster);
-
+        if (isJobPending) {
+            pendingJobMasterMap.put(jobId, new Tuple2<>(PendingSourceState.RESTORE, jobMaster));
+        } else {
+            runningJobMasterMap.put(jobId, jobMaster);
+        }
         logger.info(
                 String.format(
                         "The restore %s is in %s state, restore pipeline and take over this job running",
                         jobFullName, jobStatus));
-        CompletableFuture.runAsync(
-                () -> {
-                    try {
-                        jobMaster
-                                .getPhysicalPlan()
-                                .getPipelineList()
-                                .forEach(SubPlan::restorePipelineState);
-                        jobMaster.run();
-                    } finally {
-                        // voidCompletableFuture will be cancelled when zeta master node
-                        // shutdown to simulate master failure,
-                        // don't update runningJobMasterMap is this case.
-                        if (!jobMaster.getJobMasterCompleteFuture().isCompletedExceptionally()) {
-                            runningJobMasterMap.remove(jobId);
+        // FIFO strategy，If there is a waiting task, the subsequent task will keep waiting
+        boolean canRunJob, preApplyResources;
+        if (isJobPending) {
+            preApplyResources = jobMaster.preApplyResources();
```

Comment:
can we move all ScheduleStrategy apply resource into same place, now the different ScheduleStrategy and the places to apply for resources are not unified. When `isJobPending` is true, we apply resource in here, but when `isJobPending` is false, we apply resource in https://github.com/apache/seatunnel/pull/7693/files#diff-873557476f41c5c8e00cf79e501304f0c0709083d1d77662cacb11c9c08c201aR107